### ISSUE TITLE
More fan speeds and states

### ISSUE THREAD

**Issue by htrex** [vitorafsr/i8kutils#5]
Dear Vitor,
at first I want to say thank you for maintaining this package, it allowed me to turn the fans off on my Dell XPS 9560, it works beautifully and I'm really enjoying the silence! ;)

I wrote a post on Reddit to share how I configured it on my laptop, as it took sometime to understand what to put where, it's here https://www.reddit.com/r/Dell/comments/5y3rii/xps_9560_battery_life_optimization_and_fan/

One thing that I've noticed is that i8k supports 4 states and 3 fan speeds that on the XPS 9560 are about 0, 2400 and 4800 rpm. On Windows I've seen more states and 7 fan speeds at 0, 2400, 3100, 3700, 4100, 4800, 5100 rpm, probably a lot more recent Dell laptops support this improved scheme.

Would it be possible to enhance the fan control granularity supporting more states and fans speed?

**Comment by vitorafsr** [comment]
Dear htrex,

It is nice to maintain i8kutils for the community so that all that buy a
Dell and decide to use Linux can have its fan working as expected.

But unfortunately Dell does not contribute to this project. Never
contributed. And enhancements in more speeds although known to exist is not
achievable through code as developers do not know how to proceed.

With that view, i8kutils remains with same feature since the beginning.


Em 8 de mar de 2017 9:38 PM, "htrex" <notifications@github.com> escreveu:

Dear Vitor,
at first I want to say thank you for maintaining this package, it allowed
me to turn the fans off on my Dell XPS 9560, it works beautifully and I'm
really enjoying the silence! ;)

I wrote a post on Reddit to share how I configured it on my laptop, as it
took sometime to understand what to put where, it's here
https://www.reddit.com/r/Dell/comments/5y3rii/xps_9560_
battery_life_optimization_and_fan/

One thing that I've noticed is that i8k supports 4 states and 3 fan speeds
that on the XPS 9560 are about 0, 2400 and 4800 rpm. On Windows I've seen
more states and 7 fan speeds at 0, 2400, 3100, 3700, 4100, 4800, 5100 rpm,
probably a lot more recent Dell laptops support this improved scheme.

Would it be possible to enhance the fan control granularity supporting more
states and fans speed?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/vitorafsr/i8kutils/issues/5>, or mute the thread

**Comment by htrex** [comment]
It would be enough to know which new BIOS register addresses are available to set the fans to the new speeds or what?
Can you please detail what's needed by Dell?

I'd like to help and promote the issue looking for support from the community and to eventually put some pressure on Dell to release the specific documentation.

Thanks again.

**Comment by vitorafsr** [comment]
It is needed the docs on how to control the fans. Being said that way is
understandable by every one, not only by the engineers.

If you would like to say more details, ask for the register addresses,
function calls, register values. Today i8kutils retrieve values from a SMM
function call. SMM is a independent mode of operation of the processor.
This may be sufficient to specify.

I tried in some places in the past to find such information but was
unsuccessful. So let's try again. Count on me!


Em 10 de mar de 2017 1:08 PM, "htrex" <notifications@github.com> escreveu:

It would be enough to know which new BIOS register addresses are available
to set the fans to the new speeds or what?
Can you please detail what's needed by Dell?

I'd like to help and promote the issue looking for support from the
community and to eventually put some pressure on Dell to release the
specific documentation.

Thanks again.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/vitorafsr/i8kutils/issues/5#issuecomment-285709659>,
or mute
the thread

**Comment by htrex** [comment]
To @vitorafsr and anyone interested.
I've posted a public request to release documentation on Reddit https://www.reddit.com/r/Dell/duplicates/5ypmqn/to_dell_project_sputnik_developers_about_managing/ and Dell Project Sputink Google+ community https://plus.google.com/u/0/117746869077591524195/posts/AygFVpctW9o.
There's definitively some interest on the community around the issue.
Please, show your support there and let's see if we manage to convince Dell to release the needed information.
Thanks

**Comment by m1guelpf** [comment]
@htrex Maybe you could start a change.org campaign...

**Comment by htrex** [comment]
More than 10.000 people is already supporting the request, please have a look at https://www.reddit.com/r/Dell/duplicates/5ypmqn/to_dell_project_sputnik_developers_about_managing/

The next step is to find the more appropriate channel to let Dell know, I'm looking for it.

**Comment by vitorafsr** [comment]
In 2014 I talked to Barton George! See the link below.

https://bartongeorge.io/about/#comment-28302

**Comment by htrex** [comment]
I guess you didn't receive the documentation at the time.
I'm confident that something changed since then and now it could be the right time for Dell to step up their support to the Linux community.

**Comment by chagara** [comment]
1+

**Comment by Skotar** [comment]
1+

**Comment by dusansusic** [comment]
+1

**Comment by mikeehun** [comment]
+1

**Comment by vemundl** [comment]
+1

**Comment by ghostface** [comment]
+1

**Comment by pixmin** [comment]
Yes, I would love to be able to run fans at a speed lower then ~2500 RPM. I would much prefer to have the fans ON quieter for longer, than noisy fans for short periods... I wonder how quiet they could run to the slowest useful speed.

**Comment by pixmin** [comment]
I've just discovered that fan speed can be set:
http://manpages.ubuntu.com/manpages/cosmic/man1/i8kmon.1.html

> The following variables set fans speeds. If these variables are set at config file, i8kmon will not probe for them. The probe moment turns on fans at max speed.
> 
>        set status(leftspeed)   "0 1000 2000 3000"
>               Speed values (rpm) of left fan at velocities 0, 1, 2, 3
> 
>        set status(rightspeed)   "0 1000 2000 3000"
>               Speed values (rpm) of right fan at velocities 0, 1, 2, 3

However, when I set them at 500 for velocity 1 they still turn at about 2500 RPM :(

Anyone has it working?

**Comment by CeruleanSky** [comment]
From what I understand the original values came from [i8kfan](https://www.diefer.de/i8kfan/i8kfan.html) which has a link to a 2001 source code zip file with FANCTRL.asm inside.

Its header reads:

```
; FANCTRL.ASM
; The fan control functions; extracted from NBFAN.MDM and NBSVC.MDM from
; the Dell Diagnostics 1052
```

Unfortunately, that implies without their help, as has been realized, the only way forward would be to do the same, with reverse engineering of the bios, utilities, or diagnostics, just like they were found the first time.

**Comment by cgoudie** [comment]
This tbh

**Comment by J-Ogden** [comment]
Hi I know this is an old thread, I did some BIOS hacking recently to remove the CFG lock, and found the attached .txt document when running a search inside the extracted BIOS files for "Fan Control", I can't decipher it yet, but I am interested to know, would it help??? - I am running an XPS 9575 and my fans run at 0rpm, 2500 rpm (fine, but not enough for demanding applications) or 5500rpm (very loud). 


[Section_PE32_image_Setup-fan speed.sct.0.0.en-US.ifr.txt](https://github.com/vitorafsr/i8kutils/files/11781235/Section_PE32_image_Setup-fan.speed.sct.0.0.en-US.ifr.txt)
